src/main/java/net/snowflake/ingest/streaming/internal/ParquetFlusher.java

```diff
@@ -216,6 +219,9 @@ private SerializationResult serializeFromJavaObjects(
     rows.forEach(parquetWriter::writeRow);
     parquetWriter.close();
 
+    this.verifyRowCounts(
+        "serializeFromJavaObjects", parquetWriter, channelsDataPerTable, rows.size());
```

Comment:
it would be good to use `rowCount` calculated here outside of `verifyRowCounts` for the check.